src/targeting.ts

```diff
@@ -72,10 +72,11 @@ export const getHighestCpmBidsFromBidPool = hook('sync', function(bidsReceived,
       let bucketBids = [];
       const bidsByBidder = groupBy(buckets[bucketKey], 'bidderCode')
       Object.keys(bidsByBidder).forEach(key => { bucketBids.push(bidsByBidder[key].reduce(winReducer)) });
-      // if adUnitBidLimit is set, pass top N number bids
-      if (adUnitBidLimit) {
+      const adUnit = auctionManager.getAdUnits().find(au => au.code === bucketKey);
+      const bidLimit = adUnit?.bidLimit || adUnitBidLimit;
```

Comment:
there's some logic one level up that says roughly "use the bid limit only if sendAllBids is enabled". With this change that's short circuited when the ad unit bid limit is set, which I don't think we want.

Since there you also know all ad units involved it makes sense IMO to calculate the bid limit per adunit there, which lets you loop through them one time instead of once per ad unit. For example the `adUnitBidLimit` parameter passed into here could be made into a map.